crates/csskit_proc_macro/src/snapshots/csskit_proc_macro__test__group_multiple_optional.snap

```diff
@@ -0,0 +1,67 @@
+---
+source: crates/csskit_proc_macro/src/test.rs
+expression: pretty
+snapshot_kind: text
+---
+::css_parse::keyword_set!(FooKeywords { Foo : "foo", });
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize), serde())]
+enum Foo {
+    Foo(::css_parse::T![Ident]),
+    Bar((Option<crate::Color>, Option<crate::Color>, ::css_parse::T![Ident])),
+}
+#[automatically_derived]
+impl<'a> ::css_parse::Peek<'a> for Foo {
+    fn peek(p: &::css_parse::Parser<'a>, c: ::css_lexer::Cursor) -> bool {
+        use ::css_parse::Peek;
+        <::css_parse::T![Ident]>::peek(p, c) || <crate::Color>::peek(p, c)
+            || <::css_parse::T![Ident]>::peek(p, c)
+    }
+}
+#[automatically_derived]
+impl<'a> ::css_parse::Parse<'a> for Foo {
+    fn parse(p: &mut ::css_parse::Parser<'a>) -> ::css_parse::Result<Self> {
+        use ::css_parse::{Parse, Peek};
+        if let Some(keyword) = p.parse_if_peek::<FooKeywords>()? {
+            use ::css_parse::Build;
+            match keyword {
+                FooKeywords::Foo(c) => {
+                    return Ok(Self::Foo(<::css_parse::T![Ident]>::build(p, c)));
+                }
+            }
+        }
+        let val0 = p.parse_if_peek::<crate::Color>()?;
+        let val1 = p.parse_if_peek::<crate::Color>()?;
+        let val2 = p.parse::<::css_parse::T![Ident]>()?;
+        let c: ::css_lexer::Cursor = val2.into();
+        if !p.eq_ignore_ascii_case(c, "bar") {
+            Err(
+                ::css_parse::diagnostics::UnexpectedIdent(
+                    p.parse_str(c).into(),
+                    c.into(),
+                ),
+            )?
+        }
+        let val = (val0, val1, val2);
+        Ok(Self::Bar(val))
+    }
+}
+#[automatically_derived]
+impl ::css_parse::ToCursors for Foo {
+    fn to_cursors(&self, s: &mut impl ::css_parse::CursorSink) {
+        match self {
+            Self::Foo(inner) => {
+                s.append(inner.into());
+            }
+            Self::Bar(inner) => {
+                if let Some(inner) = &inner.0 {
+                    ::css_parse::ToCursors::to_cursors(inner, s);
```

Comment:
I didnt touch this code, but explored it. `to_cursors` just does the `s.append(inner.into())`. Wasnt sure if there is an appetite, to replace all the `s.append(inner.into())` generated code to `::css_parse::ToCursors::to_cursors(inner, s)`, so there is a canonical place for how "to cursors" stuff is done. 

Follow up pr maybe.